evalml/data_checks/datetime_format_data_check.py

```diff
@@ -179,71 +187,78 @@ def validate(self, X, y):
             )
             return messages
 
+        # Check if the data is monotonically increasing
+        no_nan_datetime_values = datetime_values.dropna()
+        if not pd.DatetimeIndex(no_nan_datetime_values).is_monotonic_increasing:
+            messages.append(
+                DataCheckError(
+                    message="Datetime values must be sorted in ascending order.",
+                    data_check_name=self.name,
+                    message_code=DataCheckMessageCode.DATETIME_IS_NOT_MONOTONIC,
+                ).to_dict()
+            )
+
         col_name = (
             self.datetime_column if self.datetime_column != "index" else "either index"
         )
 
-        nan_columns = datetime_values.isna().any()
-        if nan_columns:
+        ww_payload = infer_frequency(
+            pd.Series(datetime_values),
+            debug=True,
+            window_length=3,
```

Comment:
We default to a window length of 5 in the regularizer, should we maintain parity?

Reply:
Yep, great idea!